src/material/button/button.scss

```diff
@@ -12,9 +12,9 @@
 
 // Only basic and stroked buttons (not raised, FABs or icon buttons) have a hover style.
 // Use the same visual treatment for hover as for focus.
-.mat-button:hover,
-.mat-stroked-button:hover {
-  .mat-button-focus-overlay {
+.mat-button,
+.mat-stroked-button {
+  &:hover:not(.mat-button-disabled) .mat-button-focus-overlay {
```

Comment:
This looks like it would increase the specificity of the style, which may cause problems. WE'll need to verify this with a TGP